testing/init.go

```diff
@@ -0,0 +1,272 @@
+package testing
+
+import (
+	"context"
+	"encoding/hex"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"net/http/httptest"
+	"os"
+	"time"
+
+	"github.com/ava-labs/avalanchego/api/metrics"
+	"github.com/ava-labs/avalanchego/database/manager"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/snow"
+	"github.com/ava-labs/avalanchego/snow/engine/common"
+	"github.com/ava-labs/avalanchego/snow/validators"
+	"github.com/ava-labs/avalanchego/utils/crypto/bls"
+	"github.com/ava-labs/avalanchego/utils/logging"
+	"github.com/ava-labs/avalanchego/utils/set"
+	avago_version "github.com/ava-labs/avalanchego/version"
+	"github.com/ava-labs/avalanchego/vms/platformvm/warp"
+	"github.com/ava-labs/hypersdk/chain"
+	"github.com/ava-labs/hypersdk/crypto/ed25519"
+	"github.com/ava-labs/hypersdk/examples/tokenvm/genesis"
+	trpc "github.com/ava-labs/hypersdk/examples/tokenvm/rpc"
+	"github.com/ava-labs/hypersdk/examples/tokenvm/utils"
+	"github.com/ava-labs/hypersdk/rpc"
+	"github.com/ava-labs/hypersdk/vm"
+	"github.com/fatih/color"
+	ginkgo "github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	"go.uber.org/zap"
+)
+
+var (
+	logFactory logging.Factory
+	log        logging.Logger
+
+	RequestTimeout time.Duration
+	Vms            int
+
+	Instances []Instance
+
+	newController func() *vm.VM
+)
+
+func init() {
+	logFactory = logging.NewFactory(logging.Config{
+		DisplayLevel: logging.Debug,
+	})
+	l, err := logFactory.Make("main")
+	if err != nil {
+		panic(err)
+	}
+	log = l
+
+	flag.DurationVar(
+		&RequestTimeout,
+		"request-timeout",
+		120*time.Second,
+		"timeout for transaction issuance and confirmation",
+	)
+	flag.IntVar(
+		&Vms,
+		"vms",
+		3,
+		"number of VMs to create",
+	)
+}
+
+type Instance struct {
+	ChainID            ids.ID
+	NodeID             ids.NodeID
+	Vm                 *vm.VM
+	ToEngine           chan common.Message
+	JSONRPCServer      *httptest.Server
+	TokenJSONRPCServer *httptest.Server
+	WebSocketServer    *httptest.Server
+	Cli                *rpc.JSONRPCClient // clients for embedded VMs
+	Tcli               *trpc.JSONRPCClient
+}
+
+func RegisterBeforeSuite(instances []Instance) {
+	_ = ginkgo.BeforeSuite(func() {
+		gomega.Expect(Vms).Should(gomega.BeNumerically(">", 1))
+
+		CreateInstances(instances)
+	})
+}
+
+func CreateInstances(instances []Instance) {
+	var err error
+	priv, err := ed25519.GeneratePrivateKey()
```

Comment:
Not all `hypervms` are guaranteed to support `ed25519`. We should allow the vm to provide a dummy genesis to the initializer.